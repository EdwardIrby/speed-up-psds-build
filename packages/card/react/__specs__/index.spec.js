import identity from 'identity-obj-proxy'
import React from 'react'
import renderer from 'react-test-renderer'

import { Card } from '../index'

test('renders required props', () => {
  const tree = renderer.create(
    <Card css={identity} image={<div />} title="The card title" />
  )
  expect(tree).toMatchSnapshot()
})

test('renders tag', () => {
  const tag = [<div>FakeSVG</div>, <span>Course</span>]
  expect(
    renderer.create(
      <Card css={identity} image={<div />} title="The card title" tag={tag} />
    )
  ).toMatchSnapshot()
})

test('renders no tag if size is small', () => {
  const tag = [<div>FakeSVG</div>, <span>Course</span>]
  expect(
    renderer.create(
      <Card
        css={identity}
        image={<div />}
        title="The card title"
        size="small"
        tag={tag}
      />
    )
  ).toMatchSnapshot()
})

test('renders fullOverlay', () => {
  expect(
    renderer.create(
      <Card
        css={identity}
        image={<div />}
        title="The card title"
        fullOverlay={<div>FakePlayBtn</div>}
      />
    )
  ).toMatchSnapshot()
})

test('renders bonusBar', () => {
  expect(
    renderer.create(
      <Card
        css={identity}
        image={<div />}
        title="The card title"
        bonusBar={<div>Anything</div>}
      />
    )
  ).toMatchSnapshot()
})

test('renders custom metadata classes', () => {
  expect(
    renderer.create(
      <Card
        css={identity}
        image={<div />}
        title="The card title"
        metadata1={[<span className="keepme">Custom</span>]}
      />
    )
  ).toMatchSnapshot()
})

test('renders link class on meta only if anchor tag', () => {
  expect(
    renderer.create(
      <Card
        css={identity}
        image={<div />}
        title="The card title"
        metadata1={[<span>No link class</span>, <a>yes link class</a>]}
      />
    )
  ).toMatchSnapshot()
})
