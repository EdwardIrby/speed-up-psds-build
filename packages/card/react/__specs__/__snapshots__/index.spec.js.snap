// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ps-card ps-card--medium"
>
  <div
    className="ps-card__image-frame"
  >
    <div
      className="ps-card__image"
    />
  </div>
  <div
    className="ps-card__title"
  >
    The card title
  </div>
</div>
`;
